### DISCUSSION TITLE
How to watch a provider without redraw ui

### DISCUSSION THREAD

**Discussion by zverbeta** [rrousselGit/riverpod#2590] [General]
Hi.

I have some doubts that I really understand what approach I should use.

I have such a case. There is a list of questions that I have to show the user one by one, when the user answers a question, it should be stored in the cache and then not shown to the user.

I think that I need to `watch` to a provider that has access to the questions answered by the user and further filter, but at the same time the interface is redrawn for me.

Any suggest?

**Comment by zverbeta** [comment]
Some examples

`final userHistoryProvider = StateNotifierProvider((ref) => HistoryStateNotifier(ref));`

The class `HistoryStateNotifier` contains a few methods that update state

```
final getFilteredUserQuestions =
    FutureProvider((ref) async {
  final userHistory = ref.watch(userHistoryProvider);
  final questions = await ref.watch(allUserQuestions.future);
  
 // Some filter logic here
 
 return filteredQuestions;
});
```

After user choose some answer, `userHistoryProvider` updated state and `getFilteredUserQuestions` called. So UI redraw and flicker.

How to cancel redraw of the current screen but keep the filtered state?

**Comment by hasanmhallak** [comment]
you can use `ref.watch(provider.select(() {}))` this way you watch the provider without rebuilding the tree.
